lib/plaindate.ts

```diff
@@ -188,8 +188,8 @@ export class PlainDate implements Temporal.PlainDate {
       this
     ));
 
-    const Duration = GetIntrinsic('%Temporal.Duration%');
-    return new Duration(years, months, weeks, days, 0, 0, 0, 0, 0, 0);
+    // const Duration = GetIntrinsic('%Temporal.Duration%')!;
+    return new Temporal.Duration(years, months, weeks, days, 0, 0, 0, 0, 0, 0);
```

Comment:
This we can't do, because it would break if user code monkey-patched `globalThis.Temporal.Duration`. (That's the reason for the GetIntrinsic mechanism appearing all over the place, so that we can continue to use the original values internally even if they are replaced in user code.) Sorry, this should've been documented somewhere more clearly! Probably in `intrinsic.ts`.

I think the best solution here (and elsewhere in the file) would be to type GetIntrinsic such that TypeScript knows that it returns Temporal.Duration for an argument of `%Temporal.Duration%`. Unfortunately I don't immediately have an idea of how to do that properly :smile:

Reply:
Could this be avoided by just doing `import { Duration } from './duration'`? Or am I missing something? 😅 
Makes sense otherwise, if importing is not sensible here I'll try to teach TypeScript how `GetIntrinsic` works 😊

Reply:
I don't think we can import as it causes circular value dependencies between the different files? If that would work that could be one potential fix.

If we need to stick with `GetIntrinsic` it should already be typed such that a key of `%Temporal.Duration%` returns `typeof Temporal.Duration`, but what probably needs to change is that the return type for GetIntrinsic needs to be `NonNullable<typeof INTRINSICS[key]>` [here](https://github.com/js-temporal/temporal-polyfill/blob/08346dc5bc809e7575eacde3200f9775fe19c378/lib/intrinsicclass.ts#L137-L139).

Reply:
Circular dependencies should be fine as there's already ecmascript -> calendar -> ecmascript :D
If just importing is an option that might be cleaner, right?

Reply:
How does TypeScript usually handle circular module dependencies?